k-distribution/k-tutorial/1_basic/03_parsing/README.md

```diff
@@ -1 +1,627 @@
+operators in different situations in order to resolve the ambiguity.
+
+For example, in C, `&&` binds tighter in precedence than `||`, meaning that
+the expression `true && false || false` would be parsed as
+`(true && false) || false`.
+
+Consider, then, the third iteration on the grammar of this definition
+(`lesson-03-e.k`):
+
+```k
+module LESSON-03-E
+
+  syntax Boolean ::= "true" | "false"
+                   | "(" Boolean ")" [bracket]
+                   > "!" Boolean [function]
+                   > Boolean "&&" Boolean [function]
+                   > Boolean "^" Boolean [function]
+                   > Boolean "||" Boolean [function]
+
+endmodule
+```
+
+In this example, some of the `|` symbols separating productions in a single
+block have been replaced with `>`. This serves to describe the
+**priority groups** associated with this block of productions.
+
+In this example, the first priority group consists of the atoms of the
+language: `true`, `false`, and the bracket operator. In general, a priority
+group starts either at the `::=` or `>` operator and extends until either the
+next `>` operator or the end of the production block. Thus, we can see that the
+second, third, fourth, and fifth priority groups in this grammar all consist
+of a single production.
+
+The meaning of these priority groups becomes apparent when parsing programs:
+A symbol with a **lesser priority**, (i.e., one that **binds looser**), cannot
+appear as the **direct child** of a symbol with a **greater priority** (i.e.,
+one that **binds tighter**. In this case, the `>` operator can be seen as a
+**less-than** operator describing a transitive partial ordering on the 
+productions in the production block, expressing their relative priority.
+
+To see this more concretely, let's look again at the program
+`true && false || false`. As noted before, previously this program was
+ambiguous because it was ambiguous whether the `&&` was a child of the `||`
+or vice versa. However, because of the rule that a symbol with lesser priority
+cannot appear as the direct child of a symbol with greater priority, and
+because we can see from the above grammar that `&&` has a greater priority than
+`||`, we **reject** the parse where `||` is under the `&&` operator. As a
+result, we are left with the unambiguous parse `(true && false) || false`.
+Similarly, `true || false && false` parses unambiguously as
+`true || (false && false)`. Conversely, if the user explicitly wants the other
+parse, they can express this parse using brackets. In other words, they can
+explicitly write the program `true && (false || false)`. This still parses
+successfully because the `||` operator is no longer the **direct** child of the
+`&&` operator, but is instead the direct child of the `()` operator, and the
+`&&` operator is an **indirect** parent. Indirect parents are completely
+ignored when disambiguating; only the direct parent/child relation matters.
```

Comment:
```suggestion
To see this more concretely, let's look again at the program
`true && false || false`. As noted before, previously this program was
ambiguous because the parser could either choose the `&&` to be a child of the `||`
or vice versa. However, because a symbol with lesser priority (in this case `||`)
cannot appear as the direct child of a symbol with greater priority (in this case `&&`),
the parser can now **reject** the parse where `||` is under the `&&` operator. As a
result, we are left with the unambiguous parse `(true && false) || false`.
Similarly, `true || false && false` parses unambiguously as
`true || (false && false)`. Conversely, if the user explicitly wants the other
parse, they can express this using brackets by explicitly writing `true && (false || false)`. This still parses
successfully because the `||` operator is no longer the **direct** child of the
`&&` operator, but is instead the direct child of the `()` operator, and the
`&&` operator is an **indirect** parent. Indirect parents are completely
ignored when disambiguating; only the direct parent/child relation matters.
```